From 9c964427656ef71210e2c5b48b279857806574e9 Mon Sep 17 00:00:00 2001
From: Francesco Giancane <francescogiancane8@gmail.com>
Date: Wed, 28 Feb 2018 21:32:24 +0100
Subject: [PATCH] Make it possible to disable the use of dlvsym() (needed for
 musl)

Add a patch to work around absence of dlvsym() on musl
(wasn't previously a problem as autotools weren't building tests by default)

Upstream-Status: Pending
Signed-off-by: Alexander Kanavin <alexander.kanavin@linux.intel.com>
Signed-off-by: Francesco Giancane <francescogiancane8@gmail.com>
---
 meson_options.txt | 3 +++
 test/meson.build  | 6 +++---
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/meson_options.txt b/meson_options.txt
index b5d7c98..aa849c2 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1,6 +1,9 @@
 option('docs',
        type: 'boolean', value: false,
        description: 'Enable generating the Epoxy API reference (depends on Doxygen)')
+option('has-dlvsym',
+       type: 'boolean', value: true,
+       description: 'Whether dlvsym() is available (it is not when using musl C library)')
 option('glx',
        type: 'combo',
        choices: [ 'auto', 'yes', 'no' ],
diff --git a/test/meson.build b/test/meson.build
index c5788b4..4c46ae6 100644
--- a/test/meson.build
+++ b/test/meson.build
@@ -92,8 +92,8 @@ if build_glx
     [ 'glx_has_extension_nocontext', [ 'glx_has_extension_nocontext.c' ], [], [], true ],
     [ 'glx_static', [ 'glx_static.c' ], [ '-DNEEDS_TO_BE_STATIC'], [ '-static' ], libtype == 'static' ],
     [ 'glx_shared_znow', [ 'glx_static.c', ], [], [ '-Wl,-z,now' ], has_znow ],
-    [ 'glx_alias_prefer_same_name', [ 'glx_alias_prefer_same_name.c', 'dlwrap.c', 'dlwrap.h' ], [], [ '-rdynamic' ], not build_apple ],
-    [ 'glx_gles2', [ 'glx_gles2.c', 'dlwrap.c', 'dlwrap.h' ], [], [ '-rdynamic' ], not build_apple ],
+    [ 'glx_alias_prefer_same_name', [ 'glx_alias_prefer_same_name.c', 'dlwrap.c', 'dlwrap.h' ], [], [ '-rdynamic' ], not build_apple and get_option('has-dlvsym') == true ],
+    [ 'glx_gles2', [ 'glx_gles2.c', 'dlwrap.c', 'dlwrap.h' ], [], [ '-rdynamic' ], not build_apple and get_option('has-dlvsym') == true ],
   ]
 
   foreach test: glx_tests
@@ -114,7 +114,7 @@ if build_glx
     endif
   endforeach
 
-  if not build_apple
+  if not build_apple and get_option('has-dlvsym') == true
     # GLX/EGL tests
     if build_egl
       glx_egl_sources = [
-- 
2.14.3

